project-koku/koku
# initial HCS report finalization

Fixes [COST-2598](https://issues.redhat.com/browse/COST-2598)

Changes proposed in this PR:
* adding initial functionality for HCS report Finalization

Testing Instructions: 
Note: make sure to watch logs, commands takea minute or two finish running
    `make docker-logs` (for koku,masu, and worker logs)
    or
    `docker logs koku_koku-worker_1 -f` just for worker logs 

1. Setup Koku dev env
`make docker-up-min-trino`
`make create-test-customer`

2. add test customer (acct10001) to unleash (login using your Red Hat email)
http://localhost:4242/#/features/strategies/hcs-data-processor
![Screen Shot 2022-05-23 at 7 35 28 AM](https://user-images.githubusercontent.com/57504257/169810729-ba2427ed-89dd-48ba-9f27-c2fa63edcab2.png)

3. Look at minIO before any data has been loaded up (let me know if you need login creds)
http://localhost:9090/buckets/koku-bucket/browse
![Screen Shot 2022-05-23 at 7 36 33 AM](https://user-images.githubusercontent.com/57504257/169810963-ca1f6fa1-46fb-4d8e-ac61-37b7522527ae.png)

4. Load up some test data (for this month and last)
`make load-test-customer-data test_source=AWS`
`make load-test-customer-data test_source=Azure`

5. Look at minIO again and pay attention to the S3 metadata (bottom right of UI)
Once you drill down to the CSV file you will see something like:
![Screen Shot 2022-05-23 at 8 09 33 AM](https://user-images.githubusercontent.com/57504257/169818101-333e2771-70b4-46e1-a3d5-d3c758cebffc.png)

6. Run masu endpoint to run report finalization
http://localhost:5042/api/cost-management/v1/hcs_report_finalization/

7. once finished you can look at minIO again  but look at previous month when drilling down to CSV file
![Screen Shot 2022-05-24 at 2 23 18 PM](https://user-images.githubusercontent.com/57504257/170105962-ce30bcfa-89c1-40e9-9f31-e3a25a7c84e0.png)
